src/Confluent.SchemaRegistry/Rest/DataContracts/Schema.cs

```diff
@@ -0,0 +1,183 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Runtime.Serialization;
+
+
+namespace Confluent.SchemaRegistry
+{
+    /// <summary>
+    ///     Represents a Schema stored in Schema Registry.
+    /// </summary>
+    [DataContract]
+    public class Schema : IComparable<Schema>, IEquatable<Schema>
+    {
+        /// <summary>
+        ///     The subject the schema is registered against.
+        /// </summary>
+        [DataMember(Name = "subject")]
+        public string Subject { get; set; }
+
+        /// <summary>
+        ///     The schema version.
+        /// </summary>
+        [DataMember(Name = "version")]
+        public int Version { get; set; }
+
+        /// <summary>
+        ///     Globally unique identifier of the schema.
+        /// </summary>
+        [DataMember(Name = "id")]
+        public int Id { get; set; }
+
+        /// <summary>
+        ///     A string representation of the schema.
+        /// </summary>
+        [DataMember(Name = "schema")]
+        public string SchemaString { get; set; }
+
+        private Schema() {}
+
+        /// <summary>
+        ///     Initializes a new instance of the Schema class.
+        /// </summary>
+        /// <param name="subject">
+        ///     The subject the schema is registered against.
+        /// </param>
+        /// <param name="version">
+        ///     The schema version, >= 0
+        /// </param>
+        /// <param name="id">
+        ///     The globally unique identifier of the schema, >= 0
+        /// </param>
+        /// <param name="schemaString">
+        ///     String representation of the schema.
+        /// </param>
+        public Schema(string subject, int version, int id, string schemaString)
+        {
+            if (string.IsNullOrEmpty(subject))
+            {
+                throw new ArgumentNullException(nameof(subject));
+            }
+            if (string.IsNullOrEmpty(schemaString))
+            {
+                throw new ArgumentNullException(nameof(schemaString));
+            }
+            if (version < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(version));
+            }
+            if (id < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(id));
+            }
+
+            Subject = subject;
+            Version = version;
+            Id = id;
+            SchemaString = schemaString;
+        }
+
+        /// <summary>
+        ///     Returns a string representation of the Schema object.
+        /// </summary>
+        /// <returns>
+        ///     A string that represents the schema object.
+        /// </returns>
+        public override string ToString()
+            => $"{{subject={Subject}, version={Version}, id={Id}, schema={SchemaString}}}";
```

Comment:
do you really want to print the full schema def here? Seems like subject, version, id are sufficient.

Reply:
yep, this is silly.